### ISSUE TITLE
Unexpected behaviour of set_process(false)

### ISSUE THREAD

**Issue by Svarshick** [godotengine/godot#94710]
### Tested versions

4.2.stable

### System information

Godot v4.2.2.stable.mono - Ubuntu Core 22 - Wayland - Vulkan (Forward+) - integrated Intel(R) Graphics (ADL GT2) - 12th Gen Intel(R) Core(TM) i5-12500H (16 Threads)

### Issue description

I found out post about the same problem: https://github.com/godotengine/godot/issues/7894
As I understood answer from [reduz](https://github.com/reduz)
```
We could add an extra check i guess
```
It was considered as bug. Let me know if not, please. 
Issue was closed as completed. 

### Steps to reproduce

I created 2D Scene (Main) and added main.gd:
```gdscript
extends Node2D

var _active = true
z
func _process(delta):
	print("_process here!")

func _enter_tree():
	set_process(_active)
	print("Parent entered!")
```
Output:
![image](https://github.com/user-attachments/assets/3e537332-919c-4dad-aef6-19eaf5510111)

### Minimal reproduction project (MRP)

[bug.zip](https://github.com/user-attachments/files/16367008/bug.zip)

**Comment by huwpascoe** [comment]
`_active` should be false for the process to be disabled?

**Comment by Svarshick** [comment]
> `_active` should be false for the process to be disabled?
Oh, I am sorry. Yes, it should be false. I may checked if it works with another parameters and forgot to change it

**Comment by huwpascoe** [comment]
Concluded testing, it only works with _ready.

```gdscript
func _init():
	set_process(false) # FAIL

func _enter_tree():
	set_process(false) # FAIL

func _ready():
	set_process(false) # OK
```

Don't know if this intended behavior or not.

**Comment by kleonc** [comment]
> I found out post about the same problem: #7894 As I understood the answer from [reduz](https://github.com/reduz)
> 
> ```
> We could add an extra check i guess
> ```
> 
> It was considered as bug. Let me know if not, please. Issue was closed as completed.

#7894 describes a different problem than what is reported in here, and #7894 seems to be already solved/fixed indeed.

This issue is rather a duplicate of #31946. Basically if `_process` etc. is present/overridden within the script, then the relevant processing is enabled internally on `NOTIFICATION_READY`, which results in overwriting any previously set value (e.g. by `set_process(_active)` within `_enter_tree` like in the MRP):
https://github.com/godotengine/godot/blob/e343dbbcc1030f04dc5833f1c19d267a17332ca9/scene/main/node.cpp#L193-L215

See also https://github.com/godotengine/godot-proposals/issues/1008.

---

> ### Tested versions
> 
> 4.2.stable
> ### System information
> 
> Godot v4.2.2.stable.mono - Ubuntu Core 22 - Wayland - Vulkan (Forward+) - integrated Intel(R) Graphics (ADL GT2) - 12th Gen Intel(R) Core(TM) i5-12500H (16 Threads)

Note that in the `4.2.2` docs (thus in the current online `stable` docs as well) of `Node.set_process` etc. the potential overwrite on `NOTIFICATION_READY` is not mentioned anyhow, as:

- In #68560 notes like `Any calls to this before [method _ready] will be ignored.` were removed, it was cherry-picked into `4.2.2`.
- New notes like `If [method _process] is overridden, this will be automatically enabled before [method _ready] is called.` were added in #86903 (`4.3-dev4+`), it was not cherry-picked to `4.2` branch.

cc @Mickeon

**Comment by Mickeon** [comment]
Yeah, there's no reason not to cherrypick the next PR, if not because other minor PRs may have made it a bit annoying to.

**Comment by notroid5** [comment]
I just ran into this after adding completely unrelated code before calling `set_process(false)` in `_init()`
If I use `process_mode = PROCESS_MODE_DISABLED` instead, it works correctly but this should not be necessary.
There seems to be a side effect bug somewhere.
(Don't have an MRP but can share the project)
